pygac/reader.py

```diff
@@ -743,15 +743,24 @@ def get_angles(self):
         """
         self.get_times()
         self.get_lonlat()
-        tle1, tle2 = self.get_tle_lines()
-        orb = Orbital(self.spacecrafts_orbital[self.spacecraft_id],
-                      line1=tle1, line2=tle2)
 
-        sat_azi, sat_elev = orb.get_observer_look(self.times[:, np.newaxis],
-                                                  self.lons, self.lats, 0)
+        try:
+            tle1, tle2 = self.get_tle_lines()
+            orb = Orbital(self.spacecrafts_orbital[self.spacecraft_id],
+                          line1=tle1, line2=tle2)
+            sat_azi, sat_elev = orb.get_observer_look(self.times[:, np.newaxis],
+                                                      self.lons, self.lats, 0)
+        except IndexError:
+            from pyorbital.orbital import get_observer_look as get_observer_look_no_tle
+            sat_alt = 850.0  # km  TIROS-N OSCAR
+            sat_azi, sat_elev = get_observer_look_no_tle(
+                self.lons[:, 204][:, np.newaxis],
+                self.lats[:, 204][:, np.newaxis],  # approximate satellite position
+                sat_alt,  # approximate satellite altitude
+                self.times[:, np.newaxis],
+                self.lons, self.lats, 0)
```

Comment:
Should we take into account the static attitude shifts from some satellites for this?

Reply:
I do not know what the  static attitude shifts is. Something included in the data? And I guess the sat_alt should be different for different satellites too.

Reply:
sometimes it is in the data (for klm), otherwise we have these: https://github.com/pytroll/pygac/blob/3162eddd6ad3b9285304990d99b35537bb48f9d1/pygac/reader.py#L51-L75